Do not use 'B0' as a variable name as it conflicts with a macro in the standard
header termios.h.  See https://trac.sagemath.org/ticket/23029
diff --git a/linbox/solutions/det.h b/linbox/solutions/det.h
index f3390be..2057234 100644
--- a/linbox/solutions/det.h
+++ b/linbox/solutions/det.h
@@ -236,9 +236,9 @@ namespace LinBox
 				}
 
 				Diagonal<Field> D (diag);
-				Compose<Blackbox,Diagonal<Field> > B0 (&A, &D);
+				Compose<Blackbox,Diagonal<Field> > B_0 (&A, &D);
 				typedef Compose<Diagonal<Field>,Compose<Blackbox,Diagonal<Field> > > Blackbox1;
-				Blackbox1 B(&D, &B0);
+				Blackbox1 B(&D, &B_0);
 
 				BlackboxContainerSymmetric<Field, Blackbox1> TF (&B, F, iter);
 
diff --git a/linbox/solutions/rank.inl b/linbox/solutions/rank.inl
index fad66a8..7fa231d 100755
--- a/linbox/solutions/rank.inl
+++ b/linbox/solutions/rank.inl
@@ -142,9 +142,9 @@ namespace LinBox
 
 
 			typedef Compose<Compose<Diagonal<Field>,Blackbox >, Diagonal<Field> > BlackBox1;
-			Diagonal<Field> D0 (d1);
-			Compose<Diagonal<Field>,Blackbox > B0 (&D0, &A);
-			BlackBox1 B (&B0, &D0);
+			Diagonal<Field> D_0 (d1);
+			Compose<Diagonal<Field>,Blackbox > B_0 (&D_0, &A);
+			BlackBox1 B (&B_0, &D_0);
 
 			BlackboxContainerSymmetric<Field, BlackBox1> TF (&B, F, iter);
 			MasseyDomain<Field, BlackboxContainerSymmetric<Field, BlackBox1> > WD (&TF, M.earlyTermThreshold ());
-- 
2.8.3

